File: common/src/main/java/de/werkelmann/rekote/address/url/UrlValidBeginningAndEndRule.java
package de.werkelmann.rekote.address.url;

import de.werkelmann.rekote.address.ConnectionRule;

public class UrlValidBeginningAndEndRule implements ConnectionRule {
    @Override
    public boolean isValid(String address) {
        return !Character.isDigit(address.charAt(0)) && address.charAt(0) != '-' && address.charAt(address.length() - 1) != '-';
    }
}

File: common/src/main/kotlin/de/werkelmann/rekote/address/url/UrlValidBeginningAndEndRule.kt
package de.werkelmann.rekote.address.url

import de.werkelmann.rekote.address.ConnectionRule

class UrlValidBeginningAndEndRule : ConnectionRule {
    override fun isValid(address: String): Boolean {
        return !Character.isDigit(address[0]) && address[0] != '-' && address[address.length - 1] != '-'
    }
}
